static/js/sheetsUtils.js

```diff
@@ -0,0 +1,54 @@
+import Sefaria from "./sefaria/sefaria";
+
+export async function getSegmentObjs(refs) {
+    const segments = [];
+
+    for (const ref of refs) {
+      const text = await Sefaria.getText(ref, { stripItags: 1 });
+      const newSegments = Sefaria.makeSegments(text, false);
+      segments.push(...newSegments);
+    }
+    return segments;
+  }
+
+export async function getNormalEnRef(ref) {
+    const text = await Sefaria.getText(ref, { stripItags: 1 });
+    return text.ref;
+}
+export async function getNormalHeRef(ref) {
+    const text = await Sefaria.getText(ref, { stripItags: 1 });
+    return text.heRef;
+}
+
+function placed_segment_mapper(lang, segmented, includeNumbers, s) {
+    if (!s[lang]) {return ""}
+
+    let numStr = "";
+    if (includeNumbers) {
+        const num = (lang=="he") ? Sefaria.hebrew.encodeHebrewNumeral(s.number) : s.number;
+        numStr = "<small>(" + num + ")</small> ";
+    }
+    let str = "<span class='segment'>" + numStr + s[lang] + "</span> ";
+    if (segmented) {
+        str = "<p>" + str + "</p>";
+    }
+    str = str.replace(/(<br\/>)+/g, ' ')
+    return str;
+}
+export function segmentsToSourceText(segments, lan, segmented, includeNumbers){
```

Comment:
this function should take a parameter `ref` instead of `segmented` and `includeNumbers`. Then it can call `shouldIncludeSegmentNums` and `shouldBeSegmented` internally. These two function then don't need to be exported.